### PR TITLE
#607: Add missing Show instances

### PR DESCRIPTION
**Pull request by matt-martin**
Randomly picked up this issue as I saw it was marked as "low hanging fruit" and I wanted to get more familiar with cats (and associated FP concepts).

One thing of note: I had to add a show instance for Tuple2 to make this work.  I can't claim to be intimately familiar with Miles' Kittens project (see: https://github.com/milessabin/kittens), but it seemed like defining Show instances for Tuples and case classes would be particularly tedious and _possibly_ something that Kittens/Shapeless could make a lot easier.

Without further ado, here is my initial attempt to address issue #607 and add a Show instance for WriterT.

### CODE REVIEW

**Comment by ceedubs** [comment]
Wonderful :+1:. Thanks @matt-martin!

See #631 for some discussion regarding the tuple instance generation that you brought up.

**Comment by stew** [comment]
:+1:

**Comment by matt-martin** [comment]
Thank you @stew and @ceedubs! Appreciate the reference to #631. Definitely interested in that discussion.

File: core/src/main/scala/cats/data/WriterT.scala [hunk 1]
```diff
@@ -44,6 +44,8 @@ final case class WriterT[F[_], L, V](run: F[(L, V)]) {
 
   def reset(implicit monoidL: Monoid[L], functorF: Functor[F]): WriterT[F, L, V] =
     mapWritten(_ => monoidL.empty)
+
+  def show(implicit F: Show[F[(L, V)]]): String = F.show(run)
```
Comment (adelbertc) [inline core/src/main/scala/cats/data/WriterT.scala]: Hm we should be able to delete this line right?

Comment (ceedubs) [inline core/src/main/scala/cats/data/WriterT.scala]: @adelbertc isn't the `Show` instance delegating to this method?

Comment (matt-martin) [inline core/src/main/scala/cats/data/WriterT.scala]: Yeah, the `Show` instance is delegating to this method.  I modeled this off the change for OptionT (see: https://github.com/typelevel/cats/pull/600/files; and in particular: https://github.com/typelevel/cats/blob/6dbd4b04aff336867f563c9f6ec04fde828aa816/core/src/main/scala/cats/data/OptionT.scala#L90).

Comment (adelbertc) [inline core/src/main/scala/cats/data/WriterT.scala]: Oh I see now, I read it as `Show[(L, V)]` - ignore me :-)

**Pull request merged**